test/e2e/operator_test.go

```diff
@@ -1,26 +1,316 @@
+
+	policyClient, err := policyclientv1beta1.NewForConfig(kubeConfig)
 	if err != nil {
 		t.Fatal(err)
 	}
+
+	// test that operator namespace exists
+	t.Run("OperatorNamespace", func(t *testing.T) {
+		_, err = kubeClient.CoreV1().Namespaces().Get(operatorclient.OperatorNamespace, metav1.GetOptions{})
+		if err != nil {
+			t.Fatal(err)
+		}
+	})
+
+	// test that pdb-atlimit alert exists when there is a pdb at limit
+	t.Run("PodDisruptionBudgetAtLimit alert", func(t *testing.T) {
+		labels := map[string]string{"app": "pdbtest"}
+		name := "pdbtest-" + randSeq(5)
+
+		ns, err := createTestNamespace(kubeClient, name)
+		if err != nil {
+			t.Fatalf("could not create test namespace: %v", err)
+		}
+		defer cleanupTestObjects(kubeClient, ns.Name)
+
+		err = podCreate(t, kubeClient, name, labels)
+		if err != nil {
+			t.Fatal(err)
+		}
+		var pods *corev1.PodList
+		// Poll to see if pod is created
+		wait.Poll(time.Second*5, time.Second*300, func() (bool, error) {
+			pods, err = kubeClient.CoreV1().Pods(name).List(metav1.ListOptions{LabelSelector: "app=pdbtest"})
+			if err != nil {
+				return false, err
+			}
+			if len(pods.Items) > 0 {
+				return true, nil
+			}
+			return false, nil
+		})
+
+		err = pdbCreate(t, policyClient, name, labels)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		fmt.Println("Waiting 15s for PodDisruptionBudget to process")
+		time.Sleep(15 * time.Second)
+
+		// Now check for alert
+		wait.Poll(time.Second*5, time.Second*300, func() (bool, error) {
+			url, bearerToken, ok := locatePrometheus(t, kubeClient)
+			if !ok {
+				return false, fmt.Errorf("Could not locate prometheus deployment")
+			}
+			query := `ALERTS{alertstate="pending",alertname="PodDisruptionBudgetAtLimit",severity="warning"} == 1`
+			err := runQuery(t, kubeClient, query, name, pods.Items[0].Name, url, bearerToken, true)
+			if err != nil {
+				return false, fmt.Errorf("Waiting for prometheus: %v", err)
+			} else {
+				return true, nil
+			}
+		})
+	})
+}
+
+func pdbCreate(t *testing.T, client *policyclientv1beta1.PolicyV1beta1Client, name string, labels map[string]string) error {
+	minAvailable := intstr.FromInt(1)
+	pdb := &policyv1beta1.PodDisruptionBudget{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: policyv1beta1.PodDisruptionBudgetSpec{
+			MinAvailable: &minAvailable,
+			Selector:     &metav1.LabelSelector{MatchLabels: labels},
+		},
+	}
+	_, err := client.PodDisruptionBudgets(name).Create(pdb)
+	return err
+}
+
+func podCreate(t *testing.T, client *kubernetes.Clientset, name string, labels map[string]string) error {
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-pod",
+			Namespace: name,
+			Labels:    labels,
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{Name: "test", Image: "centos:7", Command: []string{"sh", "-c", "trap exit TERM; while true; do sleep 5; done"}},
```

Comment:
Please prefer breaking long lines on parameters for readability, e.g

```golang
{
  Name: "test", 
  Image: "centos:7", 
  Command: []string{
     "sh", 
     "-c", 
     "trap exit TERM; while true; do sleep 5; done",
  },
}
```

Reply:
ok, done